### ISSUE TITLE
Indexes don't take advantage of performant structure

### ISSUE THREAD

**Issue by willscott** [ipld/go-car#285]
The common form of [Index](https://github.com/ipld/go-car/blob/3c99491a50b5a4adfbb187b4dba07bb6b37fbb7d/v2/index/index.go#L57) introduced in Car V2 is to lay out the CIDs in the car in a sorted table that can be quickly walked through with a binary search.

We should attempt to optimize not fully [copying the index into memory](https://github.com/ipld/go-car/blob/master/v2/index/indexsorted.go#L72) when loading a car, and just doing spot reads from the index. This can for large cars avoid a copy of hundreds of megs per individual block read.

**Comment by mvdan** [comment]
Dupe of https://github.com/ipld/go-car/issues/243 or https://github.com/ipld/go-car/issues/95, perhaps. We already have multiple issues that all boil down to: we should be able to consume some index formats directly from disk.

**Comment by willscott** [comment]
yep, happy to dupe, maybe this is more trying to indicate that this remains one of our expected performance bottle necks that we shouldn't lose track of

**Comment by aarshkshah1992** [comment]
Would be GREAT to have for https://github.com/filecoin-project/dagstore/pull/116.
